zio-http/src/main/scala/zhttp/service/server/content/handlers/ServerResponseHandler.scala

```diff
@@ -0,0 +1,125 @@
+package zhttp.service.server.content.handlers
+
+import io.netty.buffer.{ByteBuf, Unpooled}
+import io.netty.channel.ChannelHandler.Sharable
+import io.netty.channel.{ChannelHandlerContext, DefaultFileRegion, SimpleChannelInboundHandler}
+import io.netty.handler.codec.http._
+import zhttp.core.Util
+import zhttp.http.{HTTP_CHARSET, HttpData, Response, Status}
+import zhttp.service.server.ServerTimeGenerator
+import zhttp.service.{ChannelFuture, HttpRuntime, Server}
+import zio.stream.ZStream
+import zio.{UIO, ZIO}
+
+import java.io.File
+
+@Sharable
+private[zhttp] case class ServerResponseHandler[R](
+  runtime: HttpRuntime[R],
+  config: Server.Config[R, Throwable],
+  serverTime: ServerTimeGenerator,
+) extends SimpleChannelInboundHandler[Response](false) {
+
+  type Ctx = ChannelHandlerContext
+
+  override def channelRead0(ctx: Ctx, response: Response): Unit = {
+    implicit val iCtx: ChannelHandlerContext = ctx
+
+    response.status match {
+      case Status.NOT_FOUND =>
+        ctx.writeAndFlush(notFoundResponse)
+      case Status.OK        =>
+        ctx.write(encodeResponse(response))
+        response.data match {
+          case HttpData.BinaryStream(stream) =>
+            runtime.unsafeRun(ctx) {
+              writeStreamContent(stream)
+            }
+          case HttpData.File(file)           =>
+            unsafeWriteFileContent(file)
+
+          case _ => ctx.flush()
+        }
+
+      case _ =>
+        ctx.writeAndFlush(encodeResponse(response))
+    }
+    ()
+  }
+
+  override def exceptionCaught(ctx: Ctx, cause: Throwable): Unit = {
+    if (ctx.channel().isWritable) {
+      ctx.writeAndFlush(serverErrorResponse(cause)): Unit
+    }
+    config.error.fold(super.exceptionCaught(ctx, cause))(f => runtime.unsafeRun(ctx)(f(cause)))
+  }
```

Comment:
this should only handle internal errors. Basically for logging etc.

Reply:
done